File: app/src/main/java/me/zipi/navitotesla/model/Github.java
package me.zipi.navitotesla.model;

import com.google.gson.annotations.SerializedName;

import java.util.List;

import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@NoArgsConstructor(access = AccessLevel.PRIVATE)
public class Github {
    @NoArgsConstructor
    @AllArgsConstructor
    @Data
    public static class Release {
        Long id;
        @SerializedName("tag_name")
        String tagName;
        String name;
        List<Asset> assets;
        String body;
    }

    @NoArgsConstructor
    @AllArgsConstructor
    @Data
    public static class Asset {
        Long id;
        String name;
        @SerializedName("content_type")
        String contentType;
        @SerializedName("browser_download_url")
        String downloadUrl;
    }
}

File: app/src/main/kotlin/me/zipi/navitotesla/model/Github.kt
package me.zipi.navitotesla.model

import com.google.gson.annotations.SerializedName


@Suppress("unused")
class Github {
    data class Release(
        var id: Long? = null,

        @SerializedName("tag_name")
        var tagName: String? = null,
        var name: String? = null,
        var assets: List<Asset>? = null,
        var body: String? = null,
        @SerializedName("prerelease")
        var isPreRelease: Boolean? = null,
    )


    data class Asset(
        var id: Long? = null,
        var name: String? = null,

        @SerializedName("content_type")
        var contentType: String? = null,

        @SerializedName("browser_download_url")
        var downloadUrl: String? = null,
    )
}